skops/card/_model_card.py

```diff
@@ -487,21 +494,27 @@ def __init__(
         trusted: bool = False,
     ) -> None:
         self.model = model
-        self.model_diagram = model_diagram
         self.metadata = metadata or ModelCardData()
         self.template = template
         self.trusted = trusted
 
         self._data: dict[str, Section] = {}
         self._metrics: dict[str, str | float | int] = {}
 
-        self._populate_template()
+        self._populate_template(model_diagram=model_diagram)
 
-    def _populate_template(self):
-        """If initialized with a template, use it to populate the card."""
-        if not self.template:
-            return
+    def _populate_template(self, model_diagram: bool | str):
+        """If initialized with a template, use it to populate the card.
+
+        Parameters
+        ----------
+        model_diagram: bool or str
+            If model_diagram is truthy and str, use str as section name. If
+            model_diagram is True, use default section for default template or
+            ignore for non-default template. If model_diagram is False, don't
+            add it.
```

Comment:
I think this logic makes more sense (and update the code accordingly)? 

```suggestion
            If model_diagram is str, use it as section name. If
            model_diagram is True, use default section for default template or
            raise for non-default template. If model_diagram is False, don't
            add it.
```

Reply:
> If model_diagram is True ... raise for non-default template

That's what I started with. The problem is that if I initialize a `Card` object with a custom template, I would immediately get the error and need to also set `model_diagram=False`. I would find this very annoying as a user, since I never asked for the model diagram. I would prefer it if I could initialize the `Card` and only get the error if I explicitly call `model_card.add_model_plot` (same behavior as now).

Reply:
I could do that, but I wonder if this isn't overly complicated for this edge case. Especially if we want to allow passing arbitrary strings to `model_diagram`, having a special case for `"auto"` is a bit odd.

Reply:
I'd say using `"auto"` as a section name is not wise in a model  card anyway ;)

Reply:
Done